plugins/inventory/nmap.py

```diff
@@ -166,6 +178,19 @@ def parse(self, inventory, loader, path, cache=True):
                 cmd.append('--exclude')
                 cmd.append(','.join(self._options['exclude']))
 
+            if self._options['dns_resolve']:
+                cmd.append('-n')
+
+            if self._options['udp_scan']:
+                if not self._options['sudo']:
+                    raise AnsibleParserError('Sudo is required for this option')
```

Comment:
Is it always required? What if the user runs ansible as root (definitely not recommended, but possible)?